### ISSUE TITLE
.reverse takes time when it probably shouldn't

### ISSUE THREAD

**Issue by AlexDaniel** [rakudo/rakudo#2925]
Code:

```perl6
say (eager 1..999999)[0]; say now - INIT now # 0.15994831
```

```perl6
say (eager 1..999999).reverse[0]; say now - INIT now # 0.1600341
```

```perl6
say (eager 1..999999).reverse.reverse[0]; say now - INIT now # 0.3857775
```

```perl6
say (eager 1..999999).reverse.reverse.reverse.reverse.reverse.reverse.reverse.reverse[0]; say now - INIT now
# 1.8385966
```

I was expecting all of these to take 0.16 or less.

**Comment by AlexDaniel** [comment]
Actually, this is somewhat regression-like. I bisected it to this: (2017-03-01) https://github.com/rakudo/rakudo/commit/beda5767ba6934e84122a0c0458a53d919c695d0

Maybe one could argue that reversing anything more than once shouldn't be something we must optimize for, but it's kinda LTA still.

**Comment by lizmat** [comment]
Looked at with 2683b3d098 .  Unless someone really wants to fix this, I suggest we close this ticket.

**Comment by lizmat** [comment]
FWIW, the timings for the code examples are now: 0.1559145, 0.1750334 and 0.2996065 on the MBP of yours truly.  So, there *is* noticeable difference between them, but it is definitely less than before.

**Comment by lizmat** [comment]
Fixed test errors with 719eefdedf
